components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14723,4 +14724,87 @@ public String getUserID(String userName) throws APIManagementException {
         }
         return list;
     }
+
+    /**
+     * Add a new endpoint registry
+     *
+     * @param endpointRegistry EndpointRegistry
+     * @param tenantID  ID of the owner's tenant
+     * @return registryId
+     */
+    public String addEndpointRegistry(EndpointRegistry endpointRegistry, int tenantID) throws APIManagementException {
+        String query = SQLConstants.ADD_ENDPOINT_REGISTRY_SQL;
```

Comment:
Shouldn't we set autoCommit to false as in other places we have done writes?

Reply:
Yes. Fixed in e0c7935